events/app.py

```diff
@@ -0,0 +1,150 @@
+import os
+import logging
+import datetime
+import pymongo
+
+from collections import namedtuple
+from flask import Flask, request, Response
+
+app = Flask(__name__)
+
+
+def connect_to_mongodb():
+    """Connects to MongoDB Atlas database.
+
+    Returns events collection if connection is successful, and None otherwise.
+    """
+    mongodb_uri = os.environ.get("MONGODB_URI")
+    if mongodb_uri is None:
+        print("Alert: not able to find MONGODB_URI environmental variable, "
+              "no connection to MongoDB instance")
+        return None  # not able to find db config var
+    return pymongo.MongoClient(mongodb_uri).eventsDB.all_events
+
+
+EVENTS_COLL = connect_to_mongodb()  # None if can't connect
+
+
+@app.route('/v1/add', methods=['POST'])
+def add_event():
+    # TODO(cmei4444): Get event information from request data
+    info = request.form['info']
+    # TODO(cmei4444): Verify that user has event editing access
+    current_time = datetime.datetime.now()
+    info = build_event_info(info, current_time)
+    try:
+        event = Event(info)
+    except ValueError:      # missing or extra event attributes
+        return Response(
+            status=400,
+            response="Event info was entered incorrectly.",
+        )
+    if EVENTS_COLL is not None:
+        event.add_to_db(EVENTS_COLL)
+    else:
+        return Response(
+            status=500,
+            response="Database was undefined.",
+        )
+    return Response(
+        status=201,
+    )
+
+
+def build_event_info(info, time):
+    """Adds created_at time to event info dict."""
+    return dict(created_at=time, **info)
+
+
+@app.route('/v1/edit/<event_id>', methods=['PUT'])
+def edit_event(event_id):
+    """Edit the event with the given id."""
+    pass
+
+
+@app.route('/v1/', methods=['GET'])
+def get_all_events():
+    """Return a list of all events currently in the DB."""
+    pass
+
+
+@app.route('/v1/search', methods=['GET'])
+def search_event():
+    """Search for the event with the given name in the DB."""
+    pass
+
+
+@app.route('/v1/<event_id>', methods=['PUT'])
+def get_one_event(event_id):
+    """Retrieve one event by event_id."""
+    pass
+
+
+def equal_times(time_1, time_2):
+    """Determines if two times are equal.
+
+    Datetime objects are rounded to the nearest millisecond before comparison.
+    Used for comparing Event objects, since MongoDB truncates times to the
+    nearest millisecond.
+    """
+    time_1 = time_1.replace(microsecond=(time_1.microsecond // 1000) * 1000)
+    time_2 = time_2.replace(microsecond=(time_2.microsecond // 1000) * 1000)
+    return time_1 == time_2
+
+
+EVENT_ATTRIBUTES = [
+    'event_id',
+    'name',
+    'description',
+    'author',
+    'created_at',
+    'event_time']
+
+
+class Event(namedtuple("EventTuple", EVENT_ATTRIBUTES)):
+    # TODO(cmei4444): move Event class to a separate file for better
+    # code organization
+    def __new__(cls, info):
```

Comment:
You might want this to be `**info` if you want to be able to construct a new Event using keyword arguments, rather than needing to construct a dict. (You can always use the unpacking format to take a dict and turn it into keyword arguments.)

Reply:
Done in [d1ade21](https://github.com/knative-portability/large-events/pull/4/commits/d1ade21683ef7f0d5d8c52a230d650691c86cd9f)